esp/logging/logginglib/LogFailSafe.cpp

```diff
@@ -332,30 +338,24 @@ void CLogFailSafe::SafeRollover()
     m_Cleared.SafeRollover(m_logsdir.str(), receive.str(), NULL, RolloverExt);
 }
 
+//Only rollover the files inside the oldLogs.
+//This is called only when a log agent is starting.
 void CLogFailSafe::RollOldLogs()
 {
-    StringBuffer filesToFind;
-    filesToFind.appendf("%s*.log",m_LogType.str());
-
-    Owned<IDirectoryIterator> di = createDirectoryIterator(m_logsdir.str(), filesToFind.str());
-    ForEach (*di)
+    ForEachItemIn(i, oldLogs)
```

Comment:
Is it guaranteed that if we get here then every previously unacknowledged transaction will exist in a new file? If we process this loop before every transaction is re-recorded, we risk losing transactions in renamed files.

Reply:
All of previously unacknowledged transactions are added into a new tank file when CLogFailSafe is constructed. The CLogFailSafe is constructed when CLogThread is constructed. This part of the code will only be called in CLogThread::run(). So, it is guaranteed that we get here after every previously unacknowledged transaction exists in the new file.